pilot/pkg/security/trustdomain/bundle_test.go

```diff
@@ -44,13 +56,25 @@ func TestReplaceTrustDomainAliases(t *testing.T) {
 			principals:        []string{"cluster.local/ns/foo/sa/bar", "cluster.local/ns/yyy/sa/zzz"},
 			expect:            []string{"td1/ns/foo/sa/bar", "cluster.local/ns/foo/sa/bar", "td1/ns/yyy/sa/zzz", "cluster.local/ns/yyy/sa/zzz"},
 		},
+		{
+			name:              "One trust domain alias, principals with * as-is",
+			trustDomainBundle: NewTrustDomainBundle("td1", []string{"cluster.local"}),
+			principals:        []string{"*/ns/foo/sa/bar", "*sa/zzz", "*"},
+			expect:            []string{"*/ns/foo/sa/bar", "*sa/zzz", "*"},
+		},
 		{
 			name:              "Two trust domain aliases, two principals",
 			trustDomainBundle: NewTrustDomainBundle("td2", []string{"td1", "cluster.local"}),
 			principals:        []string{"cluster.local/ns/foo/sa/bar", "td1/ns/yyy/sa/zzz"},
 			expect: []string{"td2/ns/foo/sa/bar", "td1/ns/foo/sa/bar", "cluster.local/ns/foo/sa/bar",
 				"td2/ns/yyy/sa/zzz", "td1/ns/yyy/sa/zzz", "cluster.local/ns/yyy/sa/zzz"},
 		},
+		{
+			name:              "Two trust domain aliases with * prefix in trust domain",
+			trustDomainBundle: NewTrustDomainBundle("td2", []string{"foo-td1", "cluster.local"}),
+			principals:        []string{"*-td1/ns/foo/sa/bar"},
+			expect:            []string{"td2/ns/foo/sa/bar", "*-td1/ns/foo/sa/bar", "cluster.local/ns/foo/sa/bar"},
```

Comment:
Say we have td = `td2` and alias = `["foo-td1", "cluster.local"]`
- for principal `*-td1/ns/foo/sa/bar` we will generate the output with both "foo-td1" and "cluster.local", even "cluster.local" is not matched
- for principal `td2/ns/foo/sa/bar`. it's still not matched but we will only generate the output with `td2/ns/foo/sa/bar` which seems a little inconsistent from the first case

I'm not following the trust domain design but just feel a little bit strange here, do you know what's the rational for the inconsistent?

Reply:
The logic here is this: for principal `*-td1/ns/foo/sa/bar`, it matches against the new trust domain or one of the aliases (in this case foo-td1). `td2`, `foo-td1`, and `cluster.local` are treated the same, hence the generated principals: "td2/ns/foo/sa/bar", "*-td1/ns/foo/sa/bar", "cluster.local/ns/foo/sa/bar"

@liminw

Reply:
Yes. this is correct.